model/translator/translator.go

```diff
@@ -0,0 +1,37 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package translation
+
+import (
+	"fmt"
+)
+
+// ErrIncompatibleType details a type conversion error during translation.
+type ErrIncompatibleType struct {
```

Comment:
This can be private now, no need to have it public unless people intend to check if errors.Is

Reply:
I thought it might be useful to distinguish between actual serialization errors and the type errors. Can hide for now and expose later if needed I guess.